sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1026,8 +1026,17 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 				.dai_type = SOF_SDW_DAI_TYPE_JACK,
 				.dailink = {SDW_UNUSED_DAI_ID, SDW_JACK_IN_DAI_ID},
 			},
+			{
+				.direction = {true, false},
+				.dai_name = "cs42l43-dp6",
+				.dai_type = SOF_SDW_DAI_TYPE_AMP,
+				.dailink = {SDW_AMP_OUT_DAI_ID, SDW_UNUSED_DAI_ID},
+				.init = sof_sdw_cs42l43_spk_init,
+				.rtd_init = cs42l43_spk_rtd_init,
+				.quirk = SOF_CODEC_SPKR,
+			},
 		},
-		.dai_num = 3,
+		.dai_num = 4,
```

Comment:
we should probably use  SOF_SDW_MAX_DAI_NUM here to avoid duplication of definitions?

Reply:
Would disagree here, the codec has 4 DAIs if we used that define here and then someone makes a codec with 5 DAIs and the define gets updated this would break.

Reply:
ok, but then we should still bump SOF_SDW_MAX_DAI_NUM. There are additional dais that will be added for NDAI stuff, so rather than the strict minimum we should probably bump this MAX_DAI_NUM to 8 or something more future-proof rather then incremental additions for every new platforms.

Reply:
Yeah that seems prudent will bump to 8 in the next spin.

Reply:
IIRC, the machine driver will create dai links for all dais lists here. How does the speaker dai be optional?

Reply:
Support for that is added by this commit:

https://github.com/thesofproject/linux/pull/4848/commits/47bae326da237ca6cf0414dc5a8a823d1ff83a39

Specifically the lines:

			if (dai_info->quirk && !(dai_info->quirk & sof_sdw_quirk))
					continue;

Which will skip adding the DAI link if the DAI link quirk and the sof_sdw_quirk don't have any matching bits.

Reply:
Got it, thanks.